packages/transaction-controller/src/TransactionController.ts

```diff
@@ -3599,9 +3600,42 @@ export class TransactionController extends BaseController<
       );
     }
 
+    if (this.#checkIfSimulationRetriggerNeeded(transactionMeta)) {
```

Comment:
We still have the condition above and `onTransactionParamsUpdated` that also re-triggers a simulation, is it worth combining those for clarity?

Reply:
I've refactored that part a bit more, feedbacks appreciated!